cpu/nrf52/radio/nrf802154/nrf802154.c

```diff
@@ -120,6 +121,63 @@ static void _enable_tx(void)
     DEBUG("[nrf802154] Device state: TXIDLE\n");
 }
 
+/**
+ * @brief   Convert from dBm to the internal representation, when the
+ *          radio operates as a IEEE802.15.4 transceiver.
+ */
+static inline uint8_t _dbm_to_ieee802154_hwval(int8_t dbm)
+{
+    return ((dbm + ED_RSSIOFFS) / ED_RSSISCALE);
+}
+
+/**
+ * @brief   Convert from the internal representation to dBm, when the
+ *          radio operates as a IEEE802.15.4 transceiver.
+ */
+static inline int8_t _hwval_to_ieee802154_dbm(uint8_t hwval)
+{
+    return (ED_RSSISCALE * hwval) - ED_RSSIOFFS;
```

Comment:
This formula is only correct because [ED_RSSIOFFS](https://github.com/RIOT-OS/RIOT/blob/eafa381a06fed27bedc938c870567c68ceda84d3/cpu/nrf52/radio/nrf802154/nrf802154.c#L64) is defined as a positive number although it should be negative. I suggest to change the sign of the define and re-adjust this calculation.

Reply:
done! Please check last fixup